UnityProject/Assets/Scripts/Systems/FilthGenerator/FilthGenerator.cs

```diff
@@ -13,33 +14,38 @@ namespace Systems.FilthGenerator
 {
 	public class FilthGenerator : SubsystemBehaviour
 	{
-		private static readonly System.Random RANDOM = new System.Random();
+		private static readonly System.Random Random = new System.Random();
 		private Tilemap floorTilemap;
 		private TileChangeManager tileChangeManager;
 
+		[SerializeField] private bool generateFilthReagent = true;
+		[SerializeField, Range(0f,100f)]
+		private float filthDensityPercentage = 4f;
 		[SerializeField, Range(0f,100f)]
-		private float FilthDensityPercentage;
+		private float filthReagentChance = 35f;
 
 		[SerializeField] private List<GameObject> filthDecalsAndObjects = new List<GameObject>();
-		[SerializeField] private bool generateFilthReagent = true;
-		
+
 		public override void Initialize()
 		{
+			if (Initialized) return;
```

Comment:
I feel like it would be better just to fix the issue in the first place?

Reply:
Doesn't matter if we fix the issue or not, this kind of logic is intended still.

Reply:
Lets say that we add a function to allow for mappers to test subsystems during runtime and some subsystems should not be run twice 
or a debug command for admins to re-run some subsystems incase something goes wrong

You'll need to remind the game that some subsystems should not be run twice when initialized

Reply:
then that should be baked inside the SubsystemBehaviour , the issue is not fixing the original problem that it runs twice

Reply:
I've solved the issue for filth gen. Have fun looking at the comments because I'm not touching that with a ten-foot pole again